c7n/policymetrics.py

```diff
@@ -11,62 +11,44 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-import argparse
-from dateutil.parser import parse
 from datetime import datetime, timedelta
 import logging
-import os
 
 from c7n.credentials import SessionFactory
-from c7n.policy import load
 from c7n.resources import load_resources
 from c7n.utils import dumps
 
+log = logging.getLogger('custodian.policymetrics')
 
-def setup_parser():
-    parser = argparse.ArgumentParser()
-    parser.add_argument('-c', '--config', required=True)
-    parser.add_argument('--assume', dest="assume_role")
-    parser.add_argument('--profile')
-    parser.add_argument(
-        '--region', default=os.environ.get('AWS_DEFAULT_REGION', 'us-east-1'))
-    parser.add_argument('--start', type=parse)
-    parser.add_argument('--end', type=parse)
-    parser.add_argument('--days', type=int, default=14)
-    parser.add_argument('--period', type=int, default=60*24*24)
-    return parser
 
+def policy_metrics(options, policies):
 
-def main():
-    logging.basicConfig(level=logging.INFO)
-    logging.getLogger('botocore').setLevel(logging.WARNING)
-
-    parser = setup_parser()
-    options = parser.parse_args()
-    options.log_group = None
-    options.cache = None
-
+    # Are these lines needed?
     factory = SessionFactory(
         options.region, options.profile, options.assume_role)
-
     session = factory()
     client = session.client('cloudwatch')
 
     load_resources()
-    policies = load(options, options.config)
+    start, end = get_endpoints(options)
 
-    if options.start and options.end:
-        start = options.start
-        end = options.end
-    elif options.days:
-        end = datetime.utcnow()
-        start = end - timedelta(options.days)
     data = {}
     for p in policies:
-        logging.info('Getting %s metrics', p)
+        log.info('Getting %s metrics', p)
         data[p.name] = p.get_metrics(start, end, options.period)
+
     print dumps(data, indent=2)
 
 
-if __name__ == '__main__':
-    main()
+def get_endpoints(options):
```

Comment:
we may need to start a commands subpackage, this module is also very cli oriented (prints to stdout), it would need to return value to be general purpose.